tests/integration/suite/placement/apilevel/shared.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package quorum
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	grpcinsecure "google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	placementv1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+)
+
+var client = http.Client{
+	Timeout: 5 * time.Second,
+}
```

Comment:
Can't use that because:

1. It returns a new client on every call, so not leveraging TCP keepalives
2. It requires `t *testing.T` and not all callers have it.

Reply:
It's very important that we do not share state between our tests in this way. We also need to ensure to kill idle connection on test cleanup https://github.com/dapr/dapr/blob/master/tests/integration/framework/util/http.go#L32C1-L32C40

A single client can be created which is shared, only calling `HTTPClient` once for each test.

> It requires t *testing.T and not all callers have it.

Can't we add `client *http.Client` as a parameter to callers?

Reply:
Done